### ISSUE TITLE
Connection String Breaks Between Executions Due to .WithDataVolume()

### ISSUE THREAD

**Issue by rodrigosol** [dotnet/aspire#3669]
I am attaching a volume to PostgreSQL using the following code snippet:

```c#
var postgres = builder.AddPostgres("database")
                      .WithPgAdmin()
                      .WithDataVolume();
```

However, my service is unable to connect to the database, resulting in the following error:

```
Unhandled exception. Npgsql.PostgresException (0x80004005): 28P01: password authentication failed for user "postgres"
```

The connection succeeds when I remove the `.WithDataVolume();` line. It appears that the password used in the connection string is being generated anew with each execution, instead of using the password that was generated when the database was initially created.

**Comment by davidfowl** [comment]
Yes, this is a known issue when using volumes https://github.com/dotnet/aspire/issues/1151

cc @eerhardt

**Comment by rodrigosol** [comment]
@davidfowl  Sorry! I searched here before opening, but I did not find it. Thanks.

**Comment by davidfowl** [comment]
The workaround is to specify a password as a parameter

**Comment by rodrigosol** [comment]
@davidfowl, can you point out some examples or documentation that shows how to set the password? 

Probably it is obvious, but I am new to the .NET world and am on the learning curve getting used to Microsoft documentation. 

I really appreciate your help! Thanks

**Comment by davidfowl** [comment]
Wrote something here:

https://github.com/dotnet/docs-aspire/issues/672#issue-2242008335

**Comment by davidfowl** [comment]
We should document this as a known issue.

**Comment by acfuns** [comment]
Good job, .WithDataVolume() can pass the database account password validation properly now.

**Comment by davidfowl** [comment]
This was fixed in 8.1
